equinix/resource_metal_device.go

```diff
@@ -442,6 +458,42 @@ func shouldReinstall(_ context.Context, d *schema.ResourceDiff, meta interface{}
 	return !reinstall_config["enabled"].(bool)
 }
 
+func shouldReinstallOrAllowChanges(attribute string) customdiff.ResourceConditionFunc {
+	return func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) bool {
+		reinstall := shouldReinstall(ctx, d, meta)
+
+		if reinstall {
+			// If reinstall is true, we're _not_ going to reinstall,
+			// so we need to see if ForceNew should be disabled due
+			// to behavior settings
+			behavior, ok := d.GetOk("behavior")
+
+			// If we don't have behavior config, force new
+			if !ok {
+				return true
+			}
+
+			// We didn't get a behavior configuration
+			behavior_list, ok := behavior.([]interface{})
+			if !ok {
+				return true
+			}
+
+			behavior_config, ok := behavior_list[0].(map[string]interface{})
+
+			// We didn't get a behavior configuration
+			if !ok {
+				return true
+			}
+
+			allow_changes, ok := behavior_config["allow_changes"].([]string)
```

Comment:
`allow_changes` - one size fits all - nice.  It may be tricky to keep up with this - whenever we add new ForceNew fields (defaulted to ForceNew or dynamically determined), we'll have to be careful to check against allow_changes.

Would we apply this pattern across all resources? If so, would we create a single reusable `behavior` base schema and a `customdiff.ForceNewIf` helper?

Reply:
IMO we should wait until we're adding a `behavior` to another resource or 2 before generalizing the implementation; if we make this reusable too early we may find that it doesn't suit the real needs that come up later.